libcylinder/src/jwt/mod.rs

```diff
@@ -14,3 +14,504 @@
  * limitations under the License.
  * ------------------------------------------------------------------------------
  */
+
+//! Provides a set of APIs to both generate a JSON Web Token-like and cryptographically validate
+//! its contents.
+
+mod error;
+
+use std::collections::HashMap;
+
+use crate::{PublicKey, Signature, Signer, Verifier};
+
+pub use error::{JsonWebTokenBuildError, JsonWebTokenParseError};
+
+/// Builder for constructing the JWT string that would be included in request
+/// headers
+#[derive(Default)]
+pub struct JsonWebTokenBuilder {
+    header: HashMap<String, String>,
```

Comment:
Should this be "headers"?

Reply:
No there is only one header in the token (it is a single JSON object).  I've updated the doc comment to hopefully improve clarity.  Likewise, the doc comment for `with_claims` has been updated as well.